src/PlusDataCollection/PointGrey/vtkPlusSpinnakerVideoSource.cxx

```diff
@@ -0,0 +1,807 @@
+
+  try
+  {
+    // End acquiring images
+    this->Internal->CameraPtr->EndAcquisition();
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Failed to end recording. Exception text: " << e.what());
+    return PLUS_FAIL;
+  }
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::NotifyConfigured()
+{
+  // TODO: Implement some configuration checks here
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::NotifyConfigured()");
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::InternalUpdate()
+{
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::InternalUpdate()");
+
+  PlusStatus retVal = PLUS_FAIL;
+
+  try
+  {
+    // Retrieve next received image and ensure image completion
+    Spinnaker::ImagePtr pResultImage = this->Internal->CameraPtr->GetNextImage();
+    if (pResultImage->IsIncomplete())
+    {
+      cout << "Image incomplete with image status " << pResultImage->GetImageStatus() << "..." << endl << endl;
+    }
+
+    // Convert image from PointGrey to correct image type
+    Spinnaker::ImagePtr convertedImage;
+    if (this->PixelEncoding == RGB24)
+    {
+      convertedImage = pResultImage->Convert(Spinnaker::PixelFormat_RGB8, Spinnaker::HQ_LINEAR);
+    }
+    else if (this->PixelEncoding == BGR24)
+    {
+      convertedImage = pResultImage->Convert(Spinnaker::PixelFormat_BGR8, Spinnaker::HQ_LINEAR);
+    }
+    else if (this->PixelEncoding == MONO8)
+    {
+      convertedImage = pResultImage->Convert(Spinnaker::PixelFormat_Mono8, Spinnaker::HQ_LINEAR);
+    }
+    else
+    {
+      LOG_ERROR("Invalid image type selected.");
+      return PLUS_FAIL;
+    }
+
+    // add image to PLUS buffer
+
+    // get buffer
+    vtkPlusDataSource* videoSource(NULL);
+    if (this->GetFirstVideoSource(videoSource) != PLUS_SUCCESS)
+    {
+      LOG_WARNING("Failed to get video source in SpinnakerVideoSource");
+      return PLUS_FAIL;
+    }
+
+    // initialize if buffer is empty
+    if (videoSource->GetNumberOfItems() == 0)
+    {
+      if (this->PixelEncoding == RGB24 || this->PixelEncoding == BGR24)
+      {
+        videoSource->SetImageType(US_IMG_RGB_COLOR);
+        videoSource->SetPixelType(VTK_UNSIGNED_CHAR);
+        videoSource->SetNumberOfScalarComponents(3);
+        videoSource->SetInputFrameSize(this->FrameSize);
+      }
+      else if (this->PixelEncoding == MONO8)
+      {
+        videoSource->SetImageType(US_IMG_BRIGHTNESS);
+        videoSource->SetPixelType(VTK_UNSIGNED_CHAR);
+        videoSource->SetNumberOfScalarComponents(1);
+        videoSource->SetInputFrameSize(this->FrameSize);
+      }
+    }
+    
+    // add frame to PLUS buffer
+    if (this->PixelEncoding == RGB24 || this->PixelEncoding == BGR24)
+    {
+      retVal = videoSource->AddItem(
```

Comment:
Plus pixel encoding is RGB24, may have to convert from BGR24 (had this bug in OpenCV video device)

Reply:
Yes; however, I intended to set this device up so that PLUS could pipe BGR info through in case the consuming code was relying on OpenCV. Otherwise we'd be doing a BGR->RGB conversion only to do the opposite in the consuming code.

Reply:
I agree that that makes sense, however color in Plus is explicitly RGB. If we want to add pixel ordering options, then we'd need to extend data source configuration and handling. If you want that, please add an issue and we can plan it out.